### PR TITLE
Add rebuildregkeys.ps1 to rebuild the regkey if user missed them

### PR DESCRIPTION
**Pull request by licanhua**
I saw user reported distros are missing after Windows update. Because we don't have telemetry on this, and I have no idea how many users are experiencing this, and don't know if I need to create a new feature on this.
As a start, I created simple script to help user to rebuild the regkeys. It has a lot of restrictions:
1. only support known distros downloaded from Windows Store
2. only support WSL2
3. regkeys are hard coded. not support any customization like changing user id, changing distro to folders other than c:

### CODE REVIEW

**Comment by Biswa96** [comment]
So, the registry is now documented, right? 😎 

The registry changes assume the GNU/Linux distribution is in WSL2 with default user (UID = 1000) already configured. Will it be correct?

**Comment by licanhua** [comment]
> So, the registry is now documented, right? 😎
> 
No. Please don't think it as public API. It's subject to change

> The registry changes assume the GNU/Linux distribution is in WSL2 with default user (UID = 1000) already configured. Will it be correct?
>
Yes.

File: diagnostics/rebuildregkeys.ps1 [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+$template = @"
+[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Lxss]
+[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Lxss\{UUID}]
+"State"=dword:00000001
+"DistributionName"="{DISTRONAME}"
+"Version"=dword:00000002
+"BasePath"="{BASEPATH}"
+"Flags"=dword:0000000f
+"DefaultUid"=dword:000003e8
```
Comment (benhillis) [inline diagnostics/rebuildregkeys.ps1:93]: Safer to leave DefaultUid as 0 (root).

**Comment by benhillis** [comment]
> > So, the registry is now documented, right? 😎
 @licanhua - please add to the comment that the registry state is subject to change.

File: diagnostics/rebuildregkeys.ps1 [hunk 2]
```diff
@@ -0,0 +1,138 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root for license information.
+
+<#
+Script to rebuild the WSL2 distros regkeys when they are missing from user's registry but provided in Get-AppxPackage
```
Comment (benhillis) [inline diagnostics/rebuildregkeys.ps1:5]: Why is this WSL2 only? Can support be added for WSL1?

**Review by benhillis — CHANGES_REQUESTED** [review]
Let's fix to work with WSL1 and make it clear that the registry key state is subject to change and should not be used programatically.

File: diagnostics/rebuildregkeys.ps1 (hunk 2 above)
Comment (licanhua) [reply diagnostics/rebuildregkeys.ps1:7]: no. WSL1 is complex than this, and this script only rebuild for wsl2

**Comment by licanhua** [comment]
> Let's fix to work with WSL1 and make it clear that the registry key state is subject to change and should not be used programatically.

Added a warning message

**Review by scottmai — APPROVED** [review]
LGTM

**Pull request closed without merging**